pokemon-showdown

```diff
@@ -1,41 +1,25 @@
 #!/usr/bin/env node
 
-// Before running, ensure that the Typescript files have been compiled.
-//
-// Simply doing `require('../build')` here doesn't work because the `replace`
-// step of the build is asynchronous and we can't wait for it to finish easily
-// unless we shell out and use `execSync`. However, the heuristic of simply
-// checking for the presence of  './.sim-dist/dex' to determine whether a build
-// is required is somewhat risky as it only is valid from a clean repository -
-// if you make edits or have previous build artifacts lying around, this
-// script could potentially be missing additional sources or the compiled output
-// may be out of date.
-//
-// We're OK with risking being stale here unless we're starting the server
-// because we want to keep this script as fast as possible to be usable as a way
-// of interfacing with the simulator from non-JS languages. Otherwise we error
-// on the side of caution and run `node build` to ensure we're always running
-// with the latest code.
-var built = false;
-function build() {
-	require('child_process').execSync('node build', {stdio: 'inherit', cwd: __dirname});
-	built = true;
-}
-
-try {
-	require.resolve('./.sim-dist/dex');
-} catch (err) {
-	if (err.code !== 'MODULE_NOT_FOUND') throw err; // should never happen
-	build();
-}
-
 function readTeam(stream) {
 	return stream.readLine().then(line => {
 		if (line.startsWith('[') || line.includes('|')) return line;
 		return stream.readAll().then(all => (line + '\n' + all));
 	});
 }
 
+function registerTSNode() {
+	require('ts-node').register({project: './tsconfig.json', files: true, transpileOnly: true, transpiler: 'ts-node/transpilers/swc-experimental'});
+}
+
+// enable require()ing TypeScript files
+try {
+	registerTSNode();
+} catch (e) {
+	console.log(`Installing dependencies...`);
+	var decl = ['--decl', 'decl'].includes(process.argv[2]);
+	require('child_process').execSync('npm install' + (decl ? '' : ' --production'), {stdio: 'inherit'});
```

Comment:
```suggestion
	require('child_process').execSync('npm install --production'), {stdio: 'inherit'});
```

The `decl` option is specific to `./build` when building the NPM package; it's unnecessary here. See the rest of the conversation on building the NPM package.